File: redsheet-portal/src/app/database-models/culture-country.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { Country,  } from './'
export class CultureCountry {
    constructor(cultureCountry?: CultureCountry )  {
        let properties = [ "cultureCountryId", "isEgalitarian", "isIndividualistic", "isMonochronic", "isShortTerm", "countryId",];
        for (let property of properties)
            if (cultureCountry && cultureCountry[property])
                this[property] = cultureCountry[property];
    }
 
	cultureCountryId : number =   0 ;
 
	isEgalitarian : boolean = false ;
 
	isIndividualistic : boolean = false ;
 
	isMonochronic : boolean = false ;
 
	isShortTerm : boolean = false ;
 
    @range(0,2147483647)
	countryId : number =   undefined;
	country : Country  ;


}

File: RedSheet.DbEntities/Models/CultureCountry.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("CultureCountries", Schema = "dbo")]
	[Serializable]
	public partial class CultureCountry  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int CultureCountryId { get; set; } 
 
	[Required]
	public bool IsEgalitarian { get; set; } 
 
	[Required]
	public bool IsIndividualistic { get; set; } 
 
	[Required]
	public bool IsMonochronic { get; set; } 
 
	[Required]
	public bool IsShortTerm { get; set; } 
 
	[RelationshipTableAttribue("Countries","dbo")]
	//Relationships
	public int CountryId { get; set; } 
 
	[ForeignKey("CountryId")]
	public Country Country { get; set; } 

	public CultureCountry()
	{
    }

    }
}

